More Cocoa HAVE directives cleanup

annevk opened the pull request:
#### 0bdb0ca7bd505178b00a1d28dbf5b879afcaff48
<pre>
More Cocoa HAVE directives cleanup
<a href="https://bugs.webkit.org/show_bug.cgi?id=295192">https://bugs.webkit.org/show_bug.cgi?id=295192</a>

Reviewed by Per Arne Vollan.

These all appear to be leftovers from a time when they were targeting
specific OS versions.

* Source/WebCore/page/PageGroup.cpp:

PLATFORM(MAC) is redundant with HAVE(MEDIA_ACCESSIBILITY_FRAMEWORK).

* Source/WebCore/platform/cocoa/CoreLocationGeolocationProvider.mm:

defined(CL_HAS_RADAR_88834301) has been around for long enough now that
we don&apos;t have to mention it explicitly anymore.

Canonical link: <a href="https://commits.webkit.org/296906@main">https://commits.webkit.org/296906@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f7ed2807d3fa129887795cdc677802e14c1f7483

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109797 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29455 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19885 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/115818 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60034 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/111760 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30133 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38043 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83439 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112745 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24016 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98877 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63902 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23395 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17025 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59612 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/102288 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93386 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17068 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/118610 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/108349 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36836 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27291 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92446 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37209 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95144 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92269 "Found 1 new API test failure: /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37234 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14980 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/32675 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17743 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36731 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42201 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/132625 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36391 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35900 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39733 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38100 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebCore/html/MediaElementSession.cpp:
```diff
@@ -330,10 +330,6 @@ void MediaElementSession::isVisibleInViewportChanged()
     if (!element || element->isFullscreen() || element->isVisibleInViewport())
         m_elementIsHiddenUntilVisibleInViewport = false;
-
-#if PLATFORM(COCOA) && !HAVE(CGS_FIX_FOR_RADAR_97530095)
```
Should this be removed when `HAVE(CGS_FIX_FOR_RADAR_97530095)` is 0 on Cocoa platforms?

annevk replied:
Oh oops, thank you! I just went by the radar being long since fixed, but I did not mean to introduce a functional change. Let me take that out and perhaps follow-up on it separately at some point.

pvollan commented on Source/WebCore/platform/cocoa/CoreLocationGeolocationProvider.mm:
```diff
@@ -70,7 +70,7 @@ - (instancetype)initWithWebsiteIdentifier:(const String&)websiteIdentifier clien
     _isWaitingForAuthorization = YES;
     _mode = mode;
 
-#if USE(APPLE_INTERNAL_SDK) && HAVE(CORE_LOCATION_WEBSITE_IDENTIFIERS) && defined(CL_HAS_RADAR_88834301)
```
It looks like `CL_HAS_RADAR_88834301 ` is not defined. Can the code inside `#If` be removed, then?

annevk replied:
That's what I thought initially, but It's defined in the Cocoa framework we use that introduces this API. Maybe @cdumez should double check this though since he introduced this code.

pvollan replied:
Ah, good point! I did not realize it was defined there.

annevk replied:
@pvollan Chris confirmed out-of-band that this is okay so I think this PR is good for another round.

pvollan reviewed (approved):
R=me.

annevk committed:
More Cocoa HAVE directives cleanup
https://bugs.webkit.org/show_bug.cgi?id=295192

Reviewed by Per Arne Vollan.

These all appear to be leftovers from a time when they were targeting
specific OS versions.

* Source/WebCore/page/PageGroup.cpp:

PLATFORM(MAC) is redundant with HAVE(MEDIA_ACCESSIBILITY_FRAMEWORK).

* Source/WebCore/platform/cocoa/CoreLocationGeolocationProvider.mm:

defined(CL_HAS_RADAR_88834301) has been around for long enough now that
we don't have to mention it explicitly anymore.

Canonical link: https://commits.webkit.org/296906@main
